src/Tribe/Promoter/Connector.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+class Tribe__Promoter__Connector {
+
+	public static function base_url() {
+		$url = 'https://connector.tri.be/promoter-auth-connector/us-central1/promoterConnector/';
+
+		if ( defined( 'PROMOTER_AUTH_CONNECTOR_URL' ) ) {
+			$url = PROMOTER_AUTH_CONNECTOR_URL;
+		}
+
+		return $url;
+	}
+
+	public function authorize_with_connector( $user_id, $secret_key, $promoter_key, $license_key ) {
+		$url = self::base_url() . 'connect';
+
+		$payload = array(
+			'clientSecret' => $secret_key,
+			'licenseKey'   => $license_key,
+			'userId'       => $user_id,
+		);
+
+		// @todo Use Tribe Logger for this.
+		error_log( $url );
+
+		$token = \Firebase\JWT\JWT::encode( $payload, $promoter_key );
+
+		$args = array(
+			'body'      => array( 'token' => $token ),
+			'sslverify' => false,
+		);
+
+		// @todo Abstract this code, it's used three times in this class.
+		$response = wp_remote_post( $url, $args );
+		$code     = wp_remote_retrieve_response_code( $response );
+		$body     = wp_remote_retrieve_body( $response );
+
+		if ( is_wp_error( $response ) ) {
+			// @todo Use Tribe Logger for this.
+			error_log( $response->get_error_message() );
+			return false;
+		}
+
+		if ( $code > 299 ) {
+			// @todo Use Tribe Logger for this.
+			error_log( $body, 0 );
+			return false;
+		}
+
+		return true;
+	}
+
+	public static function authenticate_user_with_connector( $user_id ) {
+		if ( ! isset( $_SERVER['HTTP_AUTHORIZATION'] ) ) {
+			return $user_id;
+		}
+
+		$token = $_SERVER['HTTP_AUTHORIZATION'];
```

Comment:
I feel like there should be something else relied on for this than just HTTP_AUTHORIZATION, like a special Promoter-specific variable to check that it's Promoter's token? There's probably a few password protected sites like in certain staging scenarios that will kick off extra Promoter calls like this and it seems like it could be improved -- can it?

Reply:
Works for me! I'll use a request var instead.